### PR TITLE
Sync to arbor 0.5

### PR DESCRIPTION
**Pull request by thorstenhater**
Sync to arbor's master branch.

### CODE REVIEW

File: benchmarks/engines/busyring/arbor/ring.cpp
Comment (halfflat) [inline benchmarks/engines/busyring/arbor/ring.cpp:519]: Only issue is the comment here is out of date now (same as in Nora's PR).

**Review by halfflat — APPROVED** [review]
Approved, but if you could tweak that errant comment before merging...

**Comment by noraabiakar** [comment]
Since this PR is supposed to sync nsuite to Arbor master, shouldn't we also update the files in `validation/src`? I made sure these were buildable in the last update to master, and I'd rather not have them break again. Or maybe this PR should go into dev instead?

**Comment by halfflat** [comment]
In the spirit of working towards master building and running against 0.5 correctly, and then reforking dev, let's just rename the PR to something like 'Update nsuite benchmarks for Arbor 0.5', and then move on to the validation tests.

Edit: Though that does mean a period where master doesn't work correctly, I do grant. If we can get this all sorted quickly though, I think we can look the other way :)

File: benchmarks/engines/busyring/arbor/ring.cpp
Comment (thorstenhater) [reply benchmarks/engines/busyring/arbor/ring.cpp:519]: That's true. Never use comments, they lie.
I'll fix it.

**Comment by halfflat** [comment]
I'll merge, and then make a PR for the validation code, the tiny comment fix, and the 0.5 Arbor tag.

**Pull request merged**
